primitives/src/smart_extension.rs

```diff
@@ -0,0 +1,28 @@
+use codec::{Decode, Encode};
+use sp_std::prelude::Vec;
+
+/// Smart Extension types
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum SmartExtensionTypes {
+    TransferManager,
+    Offerings,
+    Custom(Vec<u8>),
+}
+
+impl Default for SmartExtensionTypes {
+    fn default() -> Self {
+        SmartExtensionTypes::Custom(b"undefined".to_vec())
+    }
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug)]
```

Comment:
Default instances for `Eq, PartialOrd, Ord` can be provided too.

Reply:
Yes, it is.